scenarioo-client/test/spec/shared/navigation/sharePage/sharePage.service.spec.ts

```diff
@@ -17,45 +17,59 @@
 
 'use strict';
 
-describe('SharePageService', function () {
+import {SharePageService} from '../../../../../app/shared/navigation/sharePage/sharePage.service';
 
-    var SharePageService;
-    var URL = 'http://www.scenarioo.org';
+describe('SharePageService', () => {
 
-    beforeEach(angular.mock.module('scenarioo.services'));
-    beforeEach(angular.mock.module('scenarioo.filters'));
-
-    beforeEach(inject(function (_SharePageService_) {
-        SharePageService = _SharePageService_;
-    }));
+    let sharePageService: SharePageService;
+    const URL = 'http://www.scenarioo.org';
 
+    beforeEach(() => {
+        sharePageService = new SharePageService();
+    });
 
-    it('is initialized with undefined values by default', function () {
-        expectBothUrlsAreUndefined();
+    it('is initialized with undefined values by default', async () => {
+        await expectBothUrlsAreUndefined();
     });
 
-    it('stores the page Url', function() {
-        SharePageService.setPageUrl(URL);
-        expect(SharePageService.getPageUrl()).toBe(URL);
+    it('stores the page Url', (done) => {
+        sharePageService.setPageUrl(URL);
+        sharePageService.getPageUrl().subscribe((value) => {
+            expect(value).toBe(URL);
+            done();
+        });
     });
 
-    it('stores the image Url', function() {
-        SharePageService.setImageUrl(URL);
-        expect(SharePageService.getImageUrl()).toBe(URL);
+    it('stores the image Url', (done) => {
+        sharePageService.setImageUrl(URL);
+        sharePageService.getImageUrl().subscribe((value) => {
+            expect(value).toBe(URL);
+            done();
+        });
     });
 
-    it('sets both URLs to undefined when the invalidateUrl method is called', function() {
-        SharePageService.setPageUrl(URL);
-        SharePageService.setImageUrl(URL);
+    it('sets both URLs to undefined when the invalidateUrl method is called', async () => {
+        sharePageService.setPageUrl(URL);
+        sharePageService.setImageUrl(URL);
 
-        SharePageService.invalidateUrls();
+        sharePageService.invalidateUrls();
 
-        expectBothUrlsAreUndefined();
+        await expectBothUrlsAreUndefined();
     });
 
-    function expectBothUrlsAreUndefined() {
-        expect(SharePageService.getPageUrl()).toBeUndefined();
-        expect(SharePageService.getImageUrl()).toBeUndefined();
+    async function expectBothUrlsAreUndefined() {
+        await new Promise((resolve) => {
+            sharePageService.getPageUrl().subscribe((value) => {
+                expect(value).toBeUndefined();
+                resolve();
+            });
+        });
+        await new Promise((resolve) => {
+            sharePageService.getImageUrl().subscribe((value) => {
+                expect(value).toBeUndefined();
+                resolve();
+            });
+        });
```

Comment:
I dont get why it was necessary to make this service return observables in this case - is it realy necessary ??

Reply:
see other comment - I think not